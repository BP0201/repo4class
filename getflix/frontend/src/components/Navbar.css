.nav-item a.active {
  color: white;
}

.nav-item a:hover {
  color: white;
}

.Getflix-logo-h1 {
  font-family: 'Bebas Neue', cursive;
}

ul.navbar-nav {
  display: flex;
  list-style-type: none;
  justify-content: space-between;
}
