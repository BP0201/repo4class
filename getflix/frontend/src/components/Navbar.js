import React, { useContext } from 'react';
import UserContext from '../UserContext';

import { NavLink } from 'react-router-dom';

import './Navbar.css'

const Navbar = ({ logout }) => {
    const { currentUser } = useContext(UserContext);

    return (
        <div className='Navbar container-fluid m-0 px-0'>
            <nav className='row navbar navbar-expand-sm navbar-dark' style={{backgroundColor: `#003366`}}>
                {currentUser ?
                <ul className='navbar-nav align-items-center'>
                    <li className='nav-item'>
                        <NavLink to="/" className="navbar-brand">
                            <h1 className='Getflix-logo-h1 my-0 mx-2'>getFlix</h1>
                        </NavLink>
                    </li>
                    {/* <li className='nav-item'>
                        <NavLink to="/search" className='nav-link'>Search</NavLink>
                    </li> */}

                    <li className='nav-item'>
                        <NavLink to="/search" className='nav-link'>Search</NavLink>
                    </li>

                    <li className='nav-item'>
                        <NavLink to="/favorites" className='nav-link'>Favorites</NavLink>
                    </li>
                    <li className='nav-item'>
                        <NavLink to="/settings" className='nav-link'>Settings</NavLink>
                    </li>
                    <li className='nav-item'>
                        <NavLink to="/logout"
                            onClick={logout}
                            data-testid="navbar-logout-link"
                            className='nav-link' end>

                            Logout

                        </NavLink>
                    </li>
                    {currentUser.is_admin ?
                    <li className='nav-item'>
                        <NavLink to="/adminportal" data-testid="navbar-ap-link" className='nav-link'>AP</NavLink>
                    </li>
                    :
                    ""
                    }
                </ul>
                :
                <ul className='navbar-nav align-items-center'>
                    <li className='nav-item'>
                        <NavLink to="/" className="navbar-brand">
                            <h1 className='Getflix-logo-h1 my-0 mx-2'>getFlix</h1>
                        </NavLink>
                    </li>
                    <li className='nav-item'>
                        <NavLink to="/login" data-testid="navbar-login-link" className='nav-link'>Login</NavLink>
                    </li>
                    <li className='nav-item'>
                        <NavLink to="/signup" className='nav-link'>Signup</NavLink>
                    </li>
                </ul>
                }
            </nav>
        </div>
    )
}

export default Navbar;