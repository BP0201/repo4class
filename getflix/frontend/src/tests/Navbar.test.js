import React from "react";
import { render, fireEvent } from '@testing-library/react';
import Navbar from '../components/Navbar';
import { MemoryRouter } from 'react-router-dom';
import { UserProvider } from './testUtils';

describe("Navbar", () => {
  it("matches snapshot when logged in", () => {
    const { asFragment, getByTestId } = render(
      <MemoryRouter>
        <UserProvider>
          <Navbar />
        </UserProvider>
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
    expect(getByTestId("navbar-logout-link")).toBeInTheDocument();
  });

  it("matches snapshot when logged out", () => {
    const { asFragment, getByTestId } = render(
      <MemoryRouter>
        <UserProvider currentUser={null} >
          <Navbar />
        </UserProvider>
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
    expect(getByTestId("navbar-login-link")).toBeInTheDocument();
  });

  it("matches snapshot for admin users", () => {
    const { asFragment, getByTestId } = render(
      <MemoryRouter>
        <UserProvider currentUser={{ username: "testAdmin", is_admin: true }} >
          <Navbar />
        </UserProvider>
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
    expect(getByTestId("navbar-ap-link")).toBeInTheDocument();
  });
});