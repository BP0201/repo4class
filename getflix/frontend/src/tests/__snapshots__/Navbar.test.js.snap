// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar matches snapshot for admin users 1`] = `
<DocumentFragment>
  <div
    class="Navbar container-fluid m-0 px-0"
  >
    <nav
      class="row navbar navbar-expand-sm navbar-dark"
      style="background-color: rgb(0, 51, 102);"
    >
      <ul
        class="navbar-nav align-items-center"
      >
        <li
          class="nav-item"
        >
          <a
            aria-current="page"
            class="navbar-brand active"
            href="/"
          >
            <h1
              class="Getflix-logo-h1 my-0 mx-2"
            >
              getFlix
            </h1>
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            class="nav-link"
            href="/search"
          >
            Search
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            class="nav-link"
            href="/favorites"
          >
            Favorites
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            class="nav-link"
            href="/settings"
          >
            Settings
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            class="nav-link"
            data-testid="navbar-logout-link"
            href="/logout"
          >
            Logout
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            class="nav-link"
            data-testid="navbar-ap-link"
            href="/adminportal"
          >
            AP
          </a>
        </li>
      </ul>
    </nav>
  </div>
</DocumentFragment>
`;

exports[`Navbar matches snapshot when logged in 1`] = `
<DocumentFragment>
  <div
    class="Navbar container-fluid m-0 px-0"
  >
    <nav
      class="row navbar navbar-expand-sm navbar-dark"
      style="background-color: rgb(0, 51, 102);"
    >
      <ul
        class="navbar-nav align-items-center"
      >
        <li
          class="nav-item"
        >
          <a
            aria-current="page"
            class="navbar-brand active"
            href="/"
          >
            <h1
              class="Getflix-logo-h1 my-0 mx-2"
            >
              getFlix
            </h1>
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            class="nav-link"
            href="/search"
          >
            Search
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            class="nav-link"
            href="/favorites"
          >
            Favorites
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            class="nav-link"
            href="/settings"
          >
            Settings
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            class="nav-link"
            data-testid="navbar-logout-link"
            href="/logout"
          >
            Logout
          </a>
        </li>
      </ul>
    </nav>
  </div>
</DocumentFragment>
`;

exports[`Navbar matches snapshot when logged out 1`] = `
<DocumentFragment>
  <div
    class="Navbar container-fluid m-0 px-0"
  >
    <nav
      class="row navbar navbar-expand-sm navbar-dark"
      style="background-color: rgb(0, 51, 102);"
    >
      <ul
        class="navbar-nav align-items-center"
      >
        <li
          class="nav-item"
        >
          <a
            aria-current="page"
            class="navbar-brand active"
            href="/"
          >
            <h1
              class="Getflix-logo-h1 my-0 mx-2"
            >
              getFlix
            </h1>
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            class="nav-link"
            data-testid="navbar-login-link"
            href="/login"
          >
            Login
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            class="nav-link"
            href="/signup"
          >
            Signup
          </a>
        </li>
      </ul>
    </nav>
  </div>
</DocumentFragment>
`;
